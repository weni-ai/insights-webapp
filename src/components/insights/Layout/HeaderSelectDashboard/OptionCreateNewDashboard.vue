<template>
  <UnnnicDropdownItem
    data-testid="option-new-dashboard"
    class="option-create-new-dashboard"
  >
    <UnnnicIcon
      data-testid="option-new-dashboard-icon"
      icon="add"
    />
    {{ $t('insights_header.add_new_dashboard') }}
  </UnnnicDropdownItem>
</template>

<script>
export default {
  name: 'OptionCreateNewDashboard',
};
</script>

<style lang="scss" scoped>
.option-create-new-dashboard {
  padding: $unnnic-spacing-xs;

  display: flex;
  align-items: center;
  justify-content: center;
  gap: $unnnic-spacing-nano;

  border-top: $unnnic-border-width-thinner solid $unnnic-color-neutral-light;

  color: $unnnic-color-neutral-darkest;
  font-family: $unnnic-font-family-secondary;
  font-size: $unnnic-font-size-body-gt;
}
</style>
