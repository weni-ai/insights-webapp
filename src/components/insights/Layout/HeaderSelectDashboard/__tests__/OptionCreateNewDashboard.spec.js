import { beforeEach, describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';

import OptionCreateNewDashboard from '@/components/insights/Layout/HeaderSelectDashboard/OptionCreateNewDashboard.vue';
import { unnnicDropdownItem } from '@weni/unnnic-system';

describe('OptionCreateNewDashboard', () => {
  let wrapper;

  const createWrapper = (props) => {
    return mount(OptionCreateNewDashboard, {
      props,
      global: {
        stubs: {
          UnnnicDropdownItem: unnnicDropdownItem,
          UnnnicIcon: true,
        },
        mocks: {
          $t: (msg) => msg,
        },
      },
    });
  };

  beforeEach(() => {
    wrapper = createWrapper({});
  });

  it('Should have an option with the title "add_new_dashboard"', async () => {
    const addNewDashboardButton = wrapper.findComponent(
      '[data-testid=option-new-dashboard]',
    );

    expect(addNewDashboardButton.exists()).toBe(true);
    expect(wrapper.html()).toContain(
      wrapper.vm.$t('insights_header.add_new_dashboard'),
    );
  });

  it('Should render the UnnnicIcon component with the correct icon', () => {
    const icon = wrapper.findComponent(
      '[data-testid=option-new-dashboard-icon]',
    );
    expect(icon.exists()).toBe(true);
    expect(icon.props('icon')).toBe('add');
  });

  it('Should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
