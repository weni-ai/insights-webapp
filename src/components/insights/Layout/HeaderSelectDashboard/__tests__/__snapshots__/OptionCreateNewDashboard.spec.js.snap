// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OptionCreateNewDashboard > Should match the snapshot 1`] = `
<a
  class="unnnic-dropdown-item unnnic--clickable option-create-new-dashboard"
  data-testid="option-new-dashboard"
  data-v-966a3d2f=""
>
  
  <span
    class="material-symbols-rounded unnnic-icon-scheme--neutral-darkest unnnic-icon-size--md"
    data-testid="option-new-dashboard-icon"
    data-v-966a3d2f=""
    data-v-9846e6db=""
    translate="no"
  >
    add
  </span>
   New dashboard
  
</a>
`;
